client/src/components/SearchForm.js

```diff
@@ -25,6 +26,8 @@ class SearchForm extends React.Component {
     this.renderSearchOptions = this.renderSearchOptions.bind(this);
     this.handleInputKeyPress = this.handleInputKeyPress.bind(this);
     this.handleRandomDropdownChange = this.handleRandomDropdownChange.bind(this);
+
+    this.history = createBrowserHistory();
```

Comment:
So, you actually wouldn't get the `history` object this way--React Router will take care of loading it into the props for you from the existing history object in the browser. (What you're doing here is literally creating a new stack of history, which will conflict with any existing history, lol.)

Ideally, this would actually interface with Redux, too--any history state changes will be integrated with the Redux store, but...uh, I never got around to implementing that even once I learned how to. (It's pretty simple, too.)

Anyways, if you can--for now, it would be better to modify the Router object as seen [here](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Router.md#router).

(If you're curious, here's what it might look like with Redux. This would go in client/index.js, I'll add it at some point.

```
  import { connectRouter, routerMiddleware } from 'connected-react-router';
  import { createBrowserHistory } from 'history'; 
  const history = createBrowserHistory();

  const store = createStore(
    connectRouter(history)(rootReducer),
    undefined,
    // ignore this line below, it's not related
    composeEnhancers(applyMiddleware(routerMiddleware(history), thunk))
  );
```

Reply:
Just saw the Navbar component and it used the HOC method. Fixed it :) Thanks!

Reply:
Although the Router object doesn't have the history prop but you were still using the withRouter HOC in the navbar. I'm assuming it works without it then?